magma/lib/magma/actions/base_action.rb

```diff
@@ -2,9 +2,10 @@
 
 class Magma
   class BaseAction
-    def initialize(project_name, action_params = {})
+    def initialize(project_name, user, action_params = {})
       @project_name = project_name
```

Comment:
You already made this extensive revision, but - since `user` is not required for most actions, would it be easier to simply merge it into `action_params` for the add_attribute action?